python/lsst/ts/wep/task/calcZernikesNeuralTask.py

```diff
@@ -0,0 +1,1964 @@
+    ) -> QTable:
+        """Create the Zernike table to store Zernike Coefficients.
+
+        Parameters
+        ----------
+        extraStamps: DonutStamps
+            The extrafocal stamps
+        intraStamps: DonutStamps
+            The intrafocal stamps
+        zkCoeffRaw: np.ndarray
+            Raw zernike coefficients from TARTS
+        zkCoeffAvg: np.ndarray
+            Averaged zernike coefficients
+
+        Returns
+        -------
+        table : `astropy.table.QTable`
+            Table with the Zernike coefficients
+        """
+        zkTable = self.initZkTable()
+        self.log.debug(
+            "Creating Zernike table: intra stamps=%d, extra stamps=%d",
+            len(intraStamps),
+            len(extraStamps),
+        )
+
+        # Add average row
+        row_data = {
+            "label": "average",
+            "used": True,
+            "intra_field_x": np.nan,
+            "intra_field_y": np.nan,
+            "extra_field_x": np.nan,
+            "extra_field_y": np.nan,
+            "intra_centroid_x": np.nan,
+            "intra_centroid_y": np.nan,
+            "extra_centroid_x": np.nan,
+            "extra_centroid_y": np.nan,
+            "intra_mag": np.nan,
+            "extra_mag": np.nan,
+            "intra_sn": np.nan,
+            "extra_sn": np.nan,
+            "intra_entropy": np.nan,
+            "extra_entropy": np.nan,
+            "intra_frac_bad_pix": np.nan,
+            "extra_frac_bad_pix": np.nan,
+            "intra_max_power_grad": np.nan,
+            "extra_max_power_grad": np.nan,
+            "intra_fx": np.nan,
+            "extra_fx": np.nan,
+            "intra_fy": np.nan,
+            "extra_fy": np.nan,
+        }
+
+        # Add Zernike coefficients
+        for i, j in enumerate(self.config.nollIndices):
+            if i < len(zkCoeffAvg):
+                row_data[f"Z{j}"] = zkCoeffAvg[i] * u.um
+            else:
+                row_data[f"Z{j}"] = 0.0 * u.um
+
+        zkTable.add_row(row_data)
+        self.log.debug("Added average row with %d Zernike terms", len(self.config.nollIndices))
+
+        # Add individual donut rows only if we have donut stamps
+        # For TARTS, we typically have one set of coefficients per exposure
+        # We'll create one row per donut if available, otherwise just the
+        # average row
+        max_stamps = max(len(extraStamps), len(intraStamps))
+        self.log.debug(
+            "Creating Zernike table: %d extra stamps, %d intra stamps, "
+            "zkCoeffRaw shape: %s",
+            len(extraStamps), len(intraStamps), zkCoeffRaw.shape
+        )
+
+        # Add individual donut rows
+        for i, (intraStamp, extraStamp) in enumerate(zip_longest(intraStamps, extraStamps)):
+            # Get the zernike coefficients for this donut
+            # For TARTS, we have individual coefficients per donut in
+            # total_zernikes
+            if zkCoeffRaw.ndim == 2 and zkCoeffRaw.shape[0] > i:
```

Comment:
Why do you need these checks at all, shouldn't zkCoeffRaw.shape[0] be the same as the min(len(intraStamps), len(extraStamps))